src/main/java/jenkins/branch/BranchBuildStrategy.java

```diff
@@ -184,6 +214,37 @@ public final boolean automaticBuild(@NonNull SCMSource source,
                                         @NonNull SCMRevision currRevision,
```

Comment:
we need to mark this version of the API as deprecated and `@Restricted(DoNotUse.class)` (or perhaps `NoExtenalUse`) so that people compiling against the updated dependency are forced to move to the new API

Reply:
The corresponded classes seem to be marked as `Deprecated` and this one which is the latest one is keeping `@Restricted(ProtectedExternally.class)` - is not this correct?